// @realmlib/resx/dist/name-map.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * A map which translates names returned by the rusted_realm extractor into
 * names which are used by @realmlib/resx.
 */
exports.NAME_MAP = {
    Failure: 'FAILURE',
    AccountList: 'ACCOUNTLIST',
    ShootAck: 'SHOOTACK',
    Ping: 'PING',
    ServerPlayerShoot: 'SERVERPLAYERSHOOT',
    RequestTrade: 'REQUESTTRADE',
    QuestObjId: 'QUESTOBJID',
    PlayerShoot: 'PLAYERSHOOT',
    ChooseName: 'CHOOSENAME',
    AcceptArenaDeath: 'ACCEPT_ARENA_DEATH',
    PetUpgradeRequest: 'PETUPGRADEREQUEST',
    Reconnect: 'RECONNECT',
    NameResult: 'NAMERESULT',
    TradeRequested: 'TRADEREQUESTED',
    TradeStart: 'TRADESTART',
    QuestFetchAsk: 'QUEST_FETCH_ASK',
    AcceptTrade: 'ACCEPTTRADE',
    Aoe: 'AOE',
    SetCondition: 'SETCONDITION',
    Text: 'TEXT',
    EditAccountList: 'EDITACCOUNTLIST',
    HatchPet: 'HATCH_PET',
    ImminentArenaWave: 'IMMINENT_ARENA_WAVE',
    QuestRedeem: 'QUEST_REDEEM',
    OtherHit: 'OTHERHIT',
    ChangeTrade: 'CHANGETRADE',
    InvSwap: 'INVSWAP',
    ShowEffect: 'SHOWEFFECT',
    QuestRedeemResponse: 'QUEST_REDEEM_RESPONSE',
    Update: 'UPDATE',
    KeyInfoResponse: 'KEY_INFO_RESPONSE',
    ArenaDeath: 'ARENA_DEATH',
    CheckCredits: 'CHECKCREDITS',
    NewTick: 'NEWTICK',
    LoginRewardMsg: 'LOGIN_REWARD_MSG',
    PetYardUpdate: 'PETYARDUPDATE',
    UseItem: 'USEITEM',
    CreateGuild: 'CREATEGUILD',
    Escape: 'ESCAPE',
    File: 'FILE',
    CreateSuccess: 'CREATE_SUCCESS',
    KeyInfoRequest: 'KEY_INFO_REQUEST',
    InvDrop: 'INVDROP',
    Notification: 'NOTIFICATION',
    PasswordPrompt: 'PASSWORD_PROMPT',
    VerifyEmail: 'VERIFY_EMAIL',
    Pic: 'PIC',
    ChangeGuildRank: 'CHANGEGUILDRANK',
    Create: 'CREATE',
    ActivePetUpdateRequest: 'ACTIVE_PET_UPDATE_REQUEST',
    CancelTrade: 'CANCELTRADE',
    ActivePetUpdate: 'ACTIVEPETUPDATE',
    TradeAccepted: 'TRADEACCEPTED',
    Damage: 'DAMAGE',
    PlayerText: 'PLAYERTEXT',
    NewAbility: 'NEW_ABILITY',
    PlaySound: 'PLAYSOUND',
    Load: 'LOAD',
    AllyShoot: 'ALLYSHOOT',
    PetChangeFormMsg: 'PET_CHANGE_FORM_MSG',
    GuildResult: 'GUILDRESULT',
    ClientStat: 'CLIENTSTAT',
    PlayerHit: 'PLAYERHIT',
    Buy: 'BUY',
    Reskin: 'RESKIN',
    Move: 'MOVE',
    GuildInvite: 'GUILDINVITE',
    Goto: 'GOTO',
    TradeDone: 'TRADEDONE',
    DeletePet: 'DELETE_PET',
    QuestFetchResponse: 'QUEST_FETCH_RESPONSE',
    UpdateAck: 'UPDATEACK',
    GotoAck: 'GOTOACK',
    InvResult: 'INVRESULT',
    InvitedToGuild: 'INVITEDTOGUILD',
    GroundDamage: 'GROUNDDAMAGE',
    Hello: 'HELLO',
    Pong: 'PONG',
    BuyResult: 'BUYRESULT',
    QuestRoomMsg: 'QUEST_ROOM_MSG',
    EnterArena: 'ENTER_ARENA',
    GlobalNotification: 'GLOBAL_NOTIFICATION',
    UsePortal: 'USEPORTAL',
    EnemyShoot: 'ENEMYSHOOT',
    TradeChanged: 'TRADECHANGED',
    JoinGuild: 'JOINGUILD',
    SquareHit: 'SQUAREHIT',
    EnemyHit: 'ENEMYHIT',
    EvolvePet: 'EVOLVE_PET',
    ReskinUnlock: 'RESKIN_UNLOCK',
    Teleport: 'TELEPORT',
    MapInfo: 'MAPINFO',
    GuildRemove: 'GUILDREMOVE',
    AoeAck: 'AOEACK',
    ClaimLoginRewardMsg: 'CLAIM_LOGIN_REWARD_MSG',
    Death: 'DEATH',
    PetChangeSkinMsg: 'PET_CHANGE_SKIN_MSG',
    RealmHeroLeftMsg: 'REALM_HERO_LEFT_MSG',
    ResetDailyQuests: 'RESET_DAILY_QUESTS',
};
